Presentation.Web/app/components/it-system/usage/tabs/it-system-usage-tab-GDPR.controller.ts

```diff
@@ -70,6 +71,33 @@
                     }
                 }
 
+                $scope.updatePersonalData = (record: Kitos.Models.ViewModel.ItSystemUsage.IPersonalDataModel) => {
+                    if (record.checked) {
+                        itSystemUsageService.patchPersonalData(itSystemUsage.id, record.value)
+                            .then(() => updatePersonalDataSelectionValue(record));
+                        return;
+                    }
+                    itSystemUsageService.removePersonalData(itSystemUsage.id, record.value)
+                        .then(() => updatePersonalDataSelectionValue(record));
+                }
+
+                function updatePersonalDataSelectionValue(selectedRecord: Kitos.Models.ViewModel.ItSystemUsage.IPersonalDataModel) {
+                    if (selectedRecord.checked) {
+                        const selectedOptions = itSystemUsage.personalData.filter(x => x === selectedRecord.value);
+                        if (selectedOptions.length === 0) {
+                            itSystemUsage.personalData.push(selectedRecord.value);
+                        }
+                        return;
+                    }
+
+                    const selectedOptions = itSystemUsage.personalData.filter(x => x === selectedRecord.value);
+                    
+                    if (selectedOptions.length === 1) {
+                        const indexOfOption = itSystemUsage.personalData.indexOf(selectedOptions[0]);
```

Comment:
if you are going to use indexOf, why not do that in line 93 and in your condition check for != -1?